stress_test_model_loan_book.R

```diff
@@ -547,10 +560,29 @@ for (i in seq(1, nrow(transition_scenarios))) {
       year == start_year,
       technology %in% technologies,
       scenario_geography == scenario_geography_filter
+    ) #%>%
+    # distinct(investor_name, portfolio_name, company_name, ald_sector, technology,
+    #          scenario_geography, year, plan_carsten, plan_sec_carsten, term,
+    #          PD_0)
+
+  financial_data_loans_pd <- financial_data_loans %>%
+    select(company_name, company_id, ald_sector, technology, pd) %>%
+    distinct(across(everything()))
+
+  plan_carsten_loanbook <- plan_carsten_loanbook %>%
+    # ADO 879: removed company id from join, but should be re-introduced later on
+    left_join(financial_data_loans_pd, by = c("company_name", "ald_sector", "technology"))
+  # TODO: what to do with entries that have NAs for pd?
+  # TODO: kick out NAs and record the diff
+  loanbook_annual_profits <- loanbook_annual_profits %>%
+    filter(!is.na(company_id))
+
+  plan_carsten_loanbook <- plan_carsten_loanbook %>%
+    select(
+      investor_name, portfolio_name, company_name, ald_sector, technology,
+      scenario_geography, year, plan_carsten, plan_sec_carsten, term, pd
     ) %>%
-    distinct(investor_name, portfolio_name, company_name, ald_sector, technology,
-             scenario_geography, year, plan_carsten, plan_sec_carsten, term,
-             PD_0)
+    distinct_all()
```

Comment:
again, i would rather suggest to add a filtering and a checking for duplicates step.

Reply:
plan_carsten turned out to produce duplicates because I did not filter on any scenario and the variable (as it is a planned variable) is of course identical across scenarios. So I added a filter for the target scenario further up an removed this